### PR TITLE
Hardening/3075 pid file

### PR DESCRIPTION
**Pull request by kzangeli**
Fixed #3075, the problem with the PID-file

### CODE REVIEW

File: src/app/contextBroker/contextBroker.cpp [hunk 1]
```diff
@@ -1283,12 +1284,36 @@ RestService restServiceCORS[] =
...
 
 
+/* ****************************************************************************
+*
+* fileExists -
+*/
+static bool fileExists(char* path)
+{
+  if (access(path, F_OK) == 0)
+    return true;
```
Comment (fgalan) [inline src/app/contextBroker/contextBroker.cpp]: Style.

```
if (...)
{
...
}
```

File: src/app/contextBroker/contextBroker.cpp [hunk 2]
```diff
@@ -1735,6 +1762,15 @@ int main(int argC, char* argV[])
...
   lmTimeFormat(0, (char*) "%Y-%m-%dT%H:%M:%S");
 
+  //
+  // NOTE: Calling '_exit()' and not 'exit()' if 'pidFile()' returns error.
+  //       The exit-function removes the PID-file and we don't want that. We want
+  //       the PID-file to remain.
+  //       Calling '_exit()' instead of 'exit()' makes sure that the exit-function is not called.
+  //
+  if ((s = pidFile()) != 0)
+    _exit(s);
```
Comment (fgalan) [inline src/app/contextBroker/contextBroker.cpp]: Another place that needs style fixing.

File: src/app/contextBroker/contextBroker.cpp (hunk 1 above)
Comment (kzangeli) [reply src/app/contextBroker/contextBroker.cpp]: Fixed in da2ec3193

File: src/app/contextBroker/contextBroker.cpp (hunk 2 above)
Comment (kzangeli) [reply src/app/contextBroker/contextBroker.cpp]: Fixed in da2ec31

**Comment by kzangeli** [comment]
Fixed a few more style issues in contextBroker.cpp

**Comment by fgalan** [comment]
LGTM

**Pull request merged**